src/Core/Metadata/Resource/ApiResourceToLegacyResourceMetadataTrait.php

```diff
@@ -79,6 +79,25 @@ private function toArray($object): array
             $arr[$this->camelCaseToSnakeCaseNameConverter->normalize(lcfirst(substr($methodName, 3)))] = $value;
         }
 
-        return $arr;
+        return $this->transformUriVariablesToIdentifiers($arr);
+    }
+
+    private function transformUriVariablesToIdentifiers(array $arrayOperation): array
+    {
+        if (isset($arrayOperation['uri_variables'])) {
```

Comment:
You can invert this condition and transform it in guard clause to reduce the complexity.